src/create_disk.sh

```diff
@@ -597,9 +608,11 @@ if [[ ${secure_execution} -eq 1 ]]; then
     if [ ! -e /dev/disk/by-id/virtio-genprotimg ]; then
         echo "Building local Secure Execution Image, running zipl and genprotimg"
         # run zipl with root hashes as kargs
+        mount -o ro,bind "/tmp/ignition.asc" "$deploy_root/etc/ignition.asc"
         rdcore_zipl_args+=("--secex-mode=enforce" "--hostkey=/dev/disk/by-id/virtio-hostkey")
         rdcore_zipl_args+=("--append-karg=rootfs.roothash=$(cat /tmp/root-roothash)")
         rdcore_zipl_args+=("--append-karg=bootfs.roothash=$(cat /tmp/boot-roothash)")
+        rdcore_zipl_args+=("--append-file=/etc/ignition.asc")
         chroot_run /usr/lib/dracut/modules.d/50rdcore/rdcore zipl "${rdcore_zipl_args[@]}"
```

Comment:
Shouldn't we delete the `/etc/ignition.asc` file here so we don't ship an empty file in the image?

Reply:
I think we had the same discussion with luks keys, but how is this going to be built in the pipeline? We don't run `rdcore zipl` there after all.

Reply:
> Shouldn't we delete the `/etc/ignition.asc` file here so we don't ship an empty file in the image?

maybe we can store this file under `/se`, will check that

Reply:
I mean we already have `rdcore_replacement`, where the files get stored for later use by the BuildVM. So if we built it into the initrd there, it should just work.
However that would write the private key to the Build LPARs disk. Unfortunately we can't really transport it in RAM to the BuildVM, as mounting a RAM Disk in a SE VM does not work.

Reply:
cannot we read this file to some `local pkey=$(< path/to/key)` and than pass it to the build VM ?

Reply:
I mean in the build we have our normal supermin vm, in which `create_disk.sh` and after that we run the SE BuildVM. The challenge is transporting the key between the 2 securely.

Reply:
I guess "build process" is trusted, or not?

Reply:
> I guess "build process" is trusted, or not?

The build environment is trusted, but unnecessarily copying around private keys isn't great either. It sounds like maybe the correct place to generate the GPG key pair is in the genprotimg VM (somewhere in `genprotimg-script.sh`?) instead where the private key can immediately be added to the initramfs before it's handed to `genprotimg`. And then for the public key, we can place it in the `/se` partition and fish it out from the `cmd-buildextend-metal` side using e.g. libguestfs?

Reply:
If we generate it in the genprotimg VM script, then i guess we can just write it to the extra disk we use to transfer kernel+initrd. We would just need to extract it before cleanup.

Reply:
That it sounds like we need 2 solutions: for local and for official builds.  Or, other way: for official builds we just store private key under `se_tmp_boot`  (== `mount /dev/disk/by-id/virtio-genprotimg ...` ) and `genprotimg-script.sh` append it to copy of `initramdisk`.  

Will implement 2nd scenario so we can discuss it

Reply:
done

Reply:
Tested with fcos. The build was successfull and i could boot the image with an encrypted ignition file.
During testing with rhcos, there were 2 things of note:
1. It needs to be RHEL 9 (RHEL8 has kernel problem with partition-label)
2. GPG version matters, encrypting the config with newer gpg fails -> Should be written in OCP docs later

Reply:
> It needs to be RHEL 9 (RHEL8 has kernel problem with partition-label)

We're planning to ship 4.13 on RHEL9, so that's OK.

> GPG version matters, encrypting the config with newer gpg fails -> Should be written in OCP docs later

That seems odd and worth digging into. What was the exact issue you hit?

Reply:
From the console:
```
[    5.646652] coreos-secex-ignition-decrypt[808]: gpg: keybox '/tmp/tmp.N4n5k40IMQ/pubring.kbx' created
[    5.677825] coreos-secex-ignition-decrypt[808]: gpg: /tmp/tmp.N4n5k40IMQ/trustdb.gpg: trustdb created
[    5.677865] coreos-secex-ignition-decrypt[808]: gpg: key BB23F4EC9B5A45E4: public key "secex" imported
[    5.686340] EXT4-fs (vda3): mounted filesystem with ordered data mode. Opts: (null)
[    5.686781] coreos-ignition-setup-user[804]: File /mnt/boot_partition/ignition/config.ign does not exist.. Skipping copy
[  OK  ] Started CoreOS Ignition User Config Setup.
[    5.736702] systemd[1]: Started CoreOS Ignition User Config Setup.
[    5.747236] coreos-secex-ignition-decrypt[808]: gpg: key BB23F4EC9B5A45E4: secret key imported
[    5.747255] coreos-secex-ignition-decrypt[808]: gpg: Total number processed: 1
[    5.747264] coreos-secex-ignition-decrypt[808]: gpg:               imported: 1
[    5.747274] coreos-secex-ignition-decrypt[808]: gpg:       secret keys read: 1
[    5.747282] coreos-secex-ignition-decrypt[808]: gpg:   secret keys imported: 1
[    5.810829] coreos-secex-ignition-decrypt[825]: gpg: [don't know]: partial length invalid for packet type 20
[    5.811759] systemd[1]: coreos-secex-ignition-decrypt.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
```
When Nikita decrypted it locally:
```
[zukku@dhcp-9-155-211-203 tmp.yC294ruiUi]$ gpg --homedir . --skip-verify --output out --decrypt cfg 
gpg: invalid armor header: hF4DpBB5SzXwUx0SAQdAqsle/FhUTes9ZKQFr4HOU1C6qtE9VDAXbDFVq2Wv+DYw\n
gpg: encrypted with cv25519 key, ID A410794B35F0531D, created 2023-02-15
      "secex"
```

Reply:
looks like this: https://www.mail-archive.com/gnupg-users@gnupg.org/msg40414.html

Reply:
Discussed this with @nikita-dubrovskii OOB. Configs encrypted using the latest GPG in e.g. Fedora are decrypted fine using the GPG in e.g. RHEL9 (I've also just verified this locally). The failure @jschintag hit above was when booting RHCOS 8.6. So this shouldn't be an issue for us.